pkg/handlers/generic/mutation/mirrors/inject.go

```diff
@@ -82,14 +84,24 @@ func (h *globalMirrorPatchHandler) Mutate(
 		v1alpha1.ImageRegistriesVariableName,
 	)
 
+	_, registryMirrorErr := variables.Get[v1alpha1.RegistryMirror](
+		vars,
+		v1alpha1.ClusterConfigVariableName,
+		[]string{"addons", v1alpha1.RegistryMirrorVariableName}...)
+
 	switch {
-	case variables.IsNotFoundError(imageRegistriesErr) && variables.IsNotFoundError(globalMirrorErr):
-		log.V(5).Info("Image Registry Credentials and Global Registry Mirror variable not defined")
+	case variables.IsNotFoundError(imageRegistriesErr) &&
+		variables.IsNotFoundError(globalMirrorErr) &&
```

Comment:
what happens in the case where global registry is provided but the distribution registry is not? 
i.e
```
globalImageRegistry:
  url: <global_registry_url>
registries:
  url: <global_registry_url>
  secretRef: <secret>
```

It should not add the containerd config for the distribution registry. It looks like this patch will add it??
Can we add a negative unit test for this?

Also I feel it could be clean to write its own patch handler for the distribution registry.

Reply:
The client will not need to know about the internals for the distribution registry like the url to use, ie. Kubernetes Service IP for the registry, is being automatically appended here to the list of mirrors.
A client can still set their own mirror if they want, and the in-cluster distribution registry mirror will be used last.

In follow up PRs the CA and credentials will also be generated at runtime in which case the client would definitely not.

Reply:
Thank you going through the code with me.